### PR TITLE
[#2376] fix(trino-connector) Gravitino-trino-connector is incompatible with Trino-435

### PR DESCRIPTION
**Pull request by diqiu50**
### What changes were proposed in this pull request?

Remove the Trino runtime dependency to avoid version incompatibility.
Our Trino-connector uses Trino-426 for development. The AllowAllAccessControl dependency, trino-toolkit-426.jar, causes compatibility issues with Trino-426

### Why are the changes needed?
Fix: #2376

### Does this PR introduce _any_ user-facing change?

NO

### How was this patch tested?

Manually testing in Trino-435

### CODE REVIEW

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/HiveConnectorAdapter.java [hunk 1]
```diff
@@ -41,6 +41,7 @@ public Map<String, Object> buildInternalConnectorConfig(GravitinoCatalog catalog
 
     Map<String, Object> properties = new HashMap<>();
     properties.put("hive.metastore.uri", catalog.getRequiredProperty("metastore.uris"));
+    properties.put("hive.security", "allow-all");
```
Comment (yuqi1129) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/HiveConnectorAdapter.java:44]: Why does only the hive need these changes?

Comment (diqiu50) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/HiveConnectorAdapter.java:44]: We change the default access Control to use  the internal connector `AccessControl` . However, by default, the Hive connector does not support the 'alter' and 'delete table' operations."

Comment (yuqi1129) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/HiveConnectorAdapter.java:44]: What is the relationship between the "Hive connector that does not support the "change" and "delete table" operations and the fact that only the Hive connector requires the modification here?

Comment (diqiu50) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/HiveConnectorAdapter.java:44]: The reference are "https://trino.io/docs/current/connector/hive-security.html#authorization"
We change the code:
```
```
to
```
  @Override
  public ConnectorAccessControl getAccessControl() {
    Connector internalConnector = catalogConnectorContext.getInternalConnector();
    return internalConnector.getAccessControl();
  }
```

**Comment by jerryshao** [comment]
@yuqi1129 please help to review it.

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Pull request merged**